### ISSUE TITLE
Something wrong with sqlite recipe

### ISSUE THREAD

**Issue by bkauler** [sabotage-linux/sabotage#439]
I have just gone through my entire build, firefox is the last one and is now compiling. Except that I had to change it back to using internal sqlite.

From previous experience, I know the flags to set so that firefox can use external sqlite, however adding those flags in the sqlite recipe does not work:
```
read CPPFLAGS << EOF
  -DPACKAGE_NAME=\"sqlite\" -DPACKAGE_TARNAME=\"sqlite\"
  -DPACKAGE_VERSION=\"3.11.1\" -DPACKAGE_STRING=\"sqlite\ 3.11.1\"
  -DPACKAGE_BUGREPORT=\"http://www.sqlite.org\"
  -DPACKAGE_URL=\"\" -DPACKAGE=\"sqlite\" -DVERSION=\"3.11.1\"
  -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1
  -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1
  -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1
  -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DHAVE_FDATASYNC=1
  -DHAVE_USLEEP=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1
  -DHAVE_DECL_STRERROR_R=1 -DHAVE_STRERROR_R=1 -DHAVE_READLINE=1
  -DHAVE_POSIX_FALLOCATE=1 -D_REENTRANT=1 -DSQLITE_THREADSAFE=1
  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE
  -DSQLITE_SECURE_DELETE -DSQLITE_ENABLE_UNLOCK_NOTIFY
EOF
```
I inserted that last line. According to firefox, sqlite was not built with those.

I am uncertain of the syntax. Is that correct to break the lines? Shouldn't those parameters all be on the same line? I know there is a "read" that will read each line.
Well, something has to be wrong.

Incidently, I think if you do it like this:

`read CPPFLAGS << 'EOF'`

Then you don't have to escape the quotes.

**Comment by bkauler** [comment]
Yep, I just did an experiment. It only assigns the first line to CPPFLAGS.

**Comment by bkauler** [comment]
This works:
```
read CPPFLAGS << 'EOF'
  -DPACKAGE_NAME="sqlite" -DPACKAGE_TARNAME="sqlite" \
  -DPACKAGE_VERSION="3.11.1" -DPACKAGE_STRING="sqlite\ 3.11.1" \
  -DPACKAGE_BUGREPORT="http://www.sqlite.org" \
  -DPACKAGE_URL="" -DPACKAGE="sqlite" -DVERSION="3.11.1" \
  -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 \
  -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 \
  -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 \
  -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DHAVE_FDATASYNC=1 \
  -DHAVE_USLEEP=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 \
  -DHAVE_DECL_STRERROR_R=1 -DHAVE_STRERROR_R=1 -DHAVE_READLINE=1 \
  -DHAVE_POSIX_FALLOCATE=1 -D_REENTRANT=1 -DSQLITE_THREADSAFE=1 \
  -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_RTREE \
  -DSQLITE_SECURE_DELETE -DSQLITE_ENABLE_UNLOCK_NOTIFY
EOF

```

**Comment by bkauler** [comment]
Um, no, EOF does not need the single quotes around it. I don't know why the double-quotes are all escaped. Can't test right now, as firefox is busy compiling, takes ages.

**Comment by rofl0r** [comment]
can we go back a step and analyse which error you were getting from firefox ? i suppose the sqlite build worked ?

**Comment by bkauler** [comment]
Firefox reported that sqlite was compiled without -DSQLITE_SECURE_DELETE, therefore the internal sqlite has to be used.

Which lead me to examine the syntax of setting CPPFLAGS in sqlite, and I found it is broken.

Note, I am using bash, with sh a symlink to bash. It could be that busybox sh does work.
But I don't see how. The 'read' will terminate on the first newline, meaning only the first line gets set to CPPFLAGS. I actually verified this with a test script. Again, my host Linux has sh symlinked to bash.
